llvm/lib/Transforms/Scalar/FPBuiltinFnSelection.cpp

```diff
@@ -154,6 +197,14 @@ static bool selectFnForFPBuiltinCalls(const TargetLibraryInfo &TLI,
     }
   }
 
+  // We don't have implementation for CUDA approximate precision builtins.
+  // Lets map them on NVPTX intrinsics. If no appropriate intrinsics are known
+  // - skip to replaceWithAltMathFunction.
+  if (T.isNVPTX() && BuiltinCall.getRequiredAccuracy().value() == 3.0) {
+    if (replaceWithNVPTXCalls(BuiltinCall))
```

Comment:
Should we somehow encode "skip"/"fallback" to alt math functions part into the name?

Reply:
Not sure, from my perspective the code here is not complex and speaks for itself :)

Reply:
My point was not in the complexity, but in the expectations. 

Let's say you read the this specific function, i.e. you start top-down to understand high-level structure first. Just by looking at the name, you may expect that this specific step will introduce some NVPTX-specific intrinsics, but in fact, it will do some other form of lowering which could be surprising for you unless you also went through the function itself.

Reply:
Sure, renamed and clarified in the comment